shared-libs/cht-datasource/src/libs/core.ts

```diff
@@ -144,3 +149,69 @@ export const getPagedGenerator = async function* <S, T>(
 
   return null;
 };
+
+/** @internal */
+export const assertTypeQualifier: (qualifier: unknown) => asserts qualifier is ContactTypeQualifier = (
+  qualifier: unknown
+) => {
+  if (!isContactTypeQualifier(qualifier)) {
+    throw new InvalidArgumentError(`Invalid contact type [${JSON.stringify(qualifier)}].`);
+  }
+};
+
+/** @internal */
+export const assertLimit: (limit: unknown) => asserts limit is number = (limit: unknown) => {
+  if (typeof limit !== 'number' || !Number.isInteger(limit) || limit <= 0) {
+    throw new InvalidArgumentError(`The limit must be a positive number: [${String(limit)}].`);
+  }
+};
+
+/** @internal */
+export const assertCursor: (cursor: unknown) => asserts cursor is Nullable<string> = (cursor: unknown) => {
+  if (cursor !== null && (typeof cursor !== 'string' || !cursor.length)) {
+    throw new InvalidArgumentError(`Invalid cursor token: [${String(cursor)}].`);
+  }
+};
+
+/** @internal */
+export const fetchAndFilter = (
+  getFunction: (limit: number, skip: number) => Promise<Nullable<Doc>[]>,
+  filterFunction: (settings: SettingsService, doc: Nullable<Doc>, uuid: string | undefined) => unknown,
+  settings: SettingsService,
+  limit: number,
+): typeof recursionInner => {
+  const recursionInner = async (
+    currentLimit: number,
+    currentSkip: number,
+    currentDocs: Nullable<Doc>[] = [],
+  ): Promise<Page<unknown>> => {
```

Comment:
Although this would eliminate the use of unknown in the function, I couldn't get this to work without the `as Page<Place.v1.Place>` assertions because `<T extends Doc>` would mean they have properties from `Doc` but `Place` and `Person` have additional properties for which errors are thrown in other parts of the code.